contrib/kind-common

```diff
@@ -320,7 +320,23 @@ is_nested_virt_enabled() {
 }
 
 install_kubevirt() {
-    local kubevirt_version="$(curl -L https://storage.googleapis.com/kubevirt-prow/release/kubevirt/kubevirt/stable.txt)"
+    # possible values:
+    # stable - install newest stable (default)
+    # vX.Y.Z - install specific stable (i.e v1.3.1)
+    # nightly - install newest nightly
+    # nightly tag - install specific nightly (i.e 20240910)
+    KUBEVIRT_VERSION=${KUBEVIRT_VERSION:-"stable"}
+
+    if [[ "$KUBEVIRT_VERSION" == "stable" ]] || [[ "$KUBEVIRT_VERSION" == v* ]]; then
+        local kubevirt_version=$( [[ "$KUBEVIRT_VERSION" == "stable" ]] && curl -sL https://storage.googleapis.com/kubevirt-prow/release/kubevirt/kubevirt/stable.txt || echo "$KUBEVIRT_VERSION" )
+        echo "Deploy stable Kubevirt release $kubevirt_version"
+        local kubevirt_release_url="https://github.com/kubevirt/kubevirt/releases/download/${kubevirt_version}"
+    else
+        local kubevirt_version=$( [[ "$KUBEVIRT_VERSION" == "nightly" ]] && curl -sL https://storage.googleapis.com/kubevirt-prow/devel/nightly/release/kubevirt/kubevirt/latest || echo "$KUBEVIRT_VERSION" )
+        echo "Deploy nightly Kubevirt build $kubevirt_version"
+        local kubevirt_release_url="https://storage.googleapis.com/kubevirt-prow/devel/nightly/release/kubevirt/kubevirt/${kubevirt_version}"
+    fi
```

Comment:
Let's just put in if statements the three use cases, even if two of the are going to have the same release URL base, it makes code cleaner and reduce inlining

```suggestion
    if [[ "$KUBEVIRT_VERSION" == "stable" ]]; then
        local kubevirt_version=$( curl -sL https://storage.googleapis.com/kubevirt-prow/release/kubevirt/kubevirt/stable.txt )
        local kubevirt_release_url="https://github.com/kubevirt/kubevirt/releases/download/${kubevirt_version}"
    elif  [[ "$KUBEVIRT_VERSION" == "v*" ]]; then
          local kubevirt_version=$KUBEVIRT_VERSION      
          local kubevirt_release_url="https://github.com/kubevirt/kubevirt/releases/download/${kubevirt_version}"
    elif [[ $KUBEVIRT_VERSION" == "nightly" ]]; then  
           local kubevirt_version=$( curl -sL https://storage.googleapis.com/kubevirt-prow/devel/nightly/release/kubevirt/kubevirt/latest )
           local kubevirt_release_url="https://storage.googleapis.com/kubevirt-prow/devel/nightly/release/kubevirt/kubevirt/${kubevirt_version}"
    else
        // error
    fi
```

Reply:
it miss one case of specific date tag, but i will try to cover it with a regex of 6 digits

Reply:
Important part is to remove inlining and create three if statements to make things clear for future selves with super dummy easy bash code.